Packs/ThousandEyes/Integrations/ThousandEyes/ThousandEyes.py

```diff
@@ -0,0 +1,758 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+# register_module_line('ThousandEyes', 'start', __line__())
+
+
+''' IMPORTS '''
+import json
+import time
+import traceback
+from datetime import datetime
+from typing import Dict, List, Optional, Tuple, Union
+
+import dateparser
+import urllib3
+from requests import Response
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+''' Global Variables '''
+INTEGRATION_NAME = 'ThousandEyes'
+INTEGRATION_COMMAND_NAME = 'thousandeyes'
+INTEGRATION_CONTEXT_NAME = 'ThousandEyes'
+API_VERSION_ENDPOINT = '/v6'
+THOUSAND_EYES_DATE_TIME_FORMAT = "%Y-%m-%dT%H:%M:%S"
+MAX_INCIDENTS_TO_FETCH = 200
+
+
+class Client(BaseClient):
+    def handle_rate_lmiit_and_make_request(
+        self,
+        method: str,
+        url_suffix: str,
+        params: Optional[Dict] = None,
+        data: Optional[Dict] = None,
+        json_data: Optional[Dict] = None,
+        timeout: float = 10,
+        resp_type: str = "response",
+        test_module: bool = False
+    ) -> Union[Response, Dict]:
+        """
+            Handles the reached rate limit, performs API request to the specified endpoint and reutrns the full Response object
+
+        Args:
+            method (str) required: The HTTP method, for example, GET, POST, and so on.
+            url_suffix (str) required: The API endpoint.
+            params (dict): URL parameters to specify the query. Default is None.
+            data (dict): The data to send in a 'POST' request. Default is None.
+            json_data (dict): The dictionary to send in a 'POST' request. Default is None.
+            timeout: (float): Time (in seconds) for the client to wait to establish a connection before a timeout occurs.
+                                Default is 10.
+            resp_type (str): Determines which data format to return from the HTTP request. Other options are 'text',
+                             'content', 'xml' or 'response'. Use 'response' to return the full response object.
+                             Default is response.
+            test_module (bool): Boolean flag to check if function is called from test-module or not
+
+        Returns:
+            Either a Response Object or Dictionary, depending on the resp_type value
+        """
+
+        if test_module:
+            response = self._http_request(
+                method=method,
+                url_suffix=url_suffix,
+                resp_type=resp_type
+            )
+
+            return response
+
+        try:
+            while True:
+                response = self._http_request(
+                    method=method,
+                    url_suffix=url_suffix,
+                    params=params,
+                    data=data,
+                    json_data=json_data,
+                    timeout=timeout,
+                    resp_type=resp_type,
+                )
+
+                response_headers = response.headers
```

Comment:
We don't have to consider that the response could be a dictionary or it's always a Response object?

Reply:
Here, the _http_request() function returns the Response object, so checking for dict instance is not required.